pkg/operator2/operator.go

```diff
@@ -361,6 +358,19 @@ func (c *authOperator) handleSync(ctx context.Context, operatorConfig *operatorv
 
 	klog.V(4).Infof("current deployment: %#v", deployment)
 
+	authConfig, err := c.authentication.Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to get auth config: %v", err)
+	}
+	routerSecret, err := c.secrets.Secrets("openshift-authentication").Get(ctx, "v4-0-config-system-router-certs", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to get router secret: %v", err)
+	}
+	ingress, err := c.ingress.Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to get ingress: %v", err)
+	}
+
 	if err := c.handleVersion(ctx, operatorConfig, authConfig, route, routerSecret, deployment, ingress, conditions); err != nil {
 		return fmt.Errorf("error checking current version: %v", err)
 	}
```

Comment:
for the below code (`handleVersion()`) - you moved metadata and route handling outside, you should include checking them (route, well-known) in the new metadata controller.

Reply:
not sure what you mean... block 2,3,4 controllers will have to do a pre-checks if the route/config/etc. exists...

i suggest they go degraded or progressing until their pre-requirements are satisfied.

Reply:
What I meant is that you moved the oauth-metadata sync code outside. This code may affect availability and correctness of the KAS /.well-known/oauth-authorization-server endpoint, therefore it would make sense to move at least
```golang
	wellknownReady, wellknownMsg, err := c.checkWellknownEndpointsReady(ctx, authConfig, route)
	conditions.handleDegraded("WellKnownEndpoint", err)
	if err != nil {
		return fmt.Errorf("unable to check the .well-known endpoint: %v", err)
	}
	if !wellknownReady {
		conditions.setProgressingTrueAndAvailableFalse("WellKnownNotReady", wellknownMsg)
		return nil
	}
```
to the new controller.

The route health-check should be moved to the same controller that handles the route as well.

Reply:
added